### ISSUE TITLE
Space in 'na me' causes missing checkboxes in plotting view

### ISSUE THREAD

**Issue by paultjevdh** [OpenModelica/OpenModelica#12235]
### Description
Consider the following model
```modelica
model test_space
Real 'a x';
Real 'b_x';
equation
der('a x')=1;
der('b_x')=-1;
end test_space;
```
The quotes around 'a x' define it as a single variable name, thus the model will compile and simulate.
However, in the plotting view only 'b_x' has checkboxes to add the signal to the plot. 'a x' cannot be selected.
![image](https://github.com/OpenModelica/OpenModelica/assets/59084059/3d8ff22c-5060-4125-b96d-b55b1c6e81a0)

The graphical model editor also rejects names with spaces, I don't know if 'a x' is invalid in modelica or not. If it is, the translator/compiler should have complained.
![image](https://github.com/OpenModelica/OpenModelica/assets/59084059/f161d157-dd8e-46e8-9a43-606db8264a15)

### Version and OS
 - OpenModelica Version 1.22.3
 - OS: Windows 11, 64 bit

**Comment by perost** [comment]
> The graphical model editor also rejects names with spaces, I don't know if 'a x' is invalid in modelica or not. If it is, the translator/compiler should have complained.

Quoted identifiers are allowed to have spaces in them, as well as commas which OMEdit also doesn't allow.

**Comment by casella** [comment]
@adeas31 I just ran your test_space MWE, opened the .mat file with MATLAB and printed the name character array (transposed); this is the outcome:
```
5×11 char array

    'time       '
    ''a x'      '
    ''b_x'      '
    'der('a x') '
    'der('b_x') '
```
as far as I can see, the information in the .mat file is written correctly. I guess the problem is in the readSimulationResultVars() API function.

**Comment by adeas31** [comment]
Or it could be the problem with our matlab reader https://github.com/OpenModelica/OpenModelica/blob/master/OMCompiler/SimulationRuntime/c/util/read_matlab4.c

**Comment by perost** [comment]
> Or it could be the problem with our matlab reader https://github.com/OpenModelica/OpenModelica/blob/master/OMCompiler/SimulationRuntime/c/util/read_matlab4.c

https://github.com/OpenModelica/OpenModelica/blob/bbff56f6c9c75b910b4f62ebefce63c04fa49a19/OMCompiler/SimulationRuntime/c/util/read_matlab4.c#L314

**Comment by casella** [comment]
I guess line 314 will have to go. Not sure why it was put there in the first place, I guess the frontend won't accept any variable name including spaces, unless it is a quoted identifier.

**Comment by AnHeuermann** [comment]
I guess we need to test even more quoted identifiers. A quoted identifier is constructed from:

>  Q-IDENT = "'" { Q-CHAR | S-ESCAPE } "'" 
>  Q-CHAR = NON-DIGIT | DIGIT | "!" | "#" | "$" | "%" | "&" | "(" | ")"
   | "*" | "+" | "," | "-" | "." | "/" | ":" | ";" | "<" | ">" | "="
   | "?" | "@" | "[" | "]" | "^" | "{" | "}" | "|" | "~" | " " | """ 
>  S-ESCAPE = "\'" | "\"" | "\?" | "\\"
   | "\a" | "\b" | "\f" | "\n" | "\r" | "\t" | "\v" 

(See https://specification.modelica.org/master/lexical-structure.html#identifiers)

A combination of `,` and `"` could break the names in CSV files.
At least it is breaking the syntax highlighting in OMEdit as well as lexing:

```modelica
model M
  Real 'a"x';
  Real 'b_x';
equation
  der('a x')=1;
  der('b_x')=-1;
end M;
```

**Comment by AnHeuermann** [comment]
@perost where is `IDENT` defined for the Lexer? I had to fix this for tree-sitter (https://github.com/OpenModelica/tree-sitter-metamodelica/commit/45c23aa3fd2d2e45ac3eb3e93744c0fe1d7fde80)

**Comment by perost** [comment]
> @perost where is `IDENT` defined for the Lexer? I had to fix this for tree-sitter ([OpenModelica/tree-sitter-metamodelica@45c23aa](https://github.com/OpenModelica/tree-sitter-metamodelica/commit/45c23aa3fd2d2e45ac3eb3e93744c0fe1d7fde80))

https://github.com/OpenModelica/OpenModelica/blob/bbff56f6c9c75b910b4f62ebefce63c04fa49a19/OMCompiler/Parser/BaseModelica_Lexer.g#L424-L435

**Comment by casella** [comment]
> I guess we need to test even more quoted identifiers

Absolutely! See, e.g., #12135. 

In fact, I recently opened OpenModelica/OpenModelicaLibraryTesting#90 which would be very useful from this point of view. @AnHeuermann would you like to help with that?

**Comment by adeas31** [comment]
Ported to maintenance/v1.23 with PR #12256

**Comment by AnHeuermann** [comment]
> In fact, I recently opened https://github.com/OpenModelica/OpenModelicaLibraryTesting/issues/90 which would be very useful from this point of view. @AnHeuermann would you like to help with that?

Yes, that would be a good idea.

**Comment by adeas31** [comment]
I closed it too quickly. The actual issue that we remove the space from the variable name is still present.
https://github.com/OpenModelica/OpenModelica/blob/bbff56f6c9c75b910b4f62ebefce63c04fa49a19/OMCompiler/SimulationRuntime/c/util/read_matlab4.c#L314

**Comment by AnHeuermann** [comment]
Oh and I need to add a test with `Real /*(y)*/ 'stupid,name';` and check if that one is escaped correctly in CSV files as well as MAT files.

**Comment by adeas31** [comment]
@sjoelund git blame shows a very old commit https://github.com/OpenModelica/OpenModelica/commit/111e61493b87281bd2626754d20c27990b8a6c9f#diff-6f508873244f3e94c3614dcb4fca9b69709980cc190c597eca9f3c4878016e8d

**Comment by sjoelund** [comment]
If that's the git blame, you need to look at commits before this. The janitor only cleaned up whitespace/etc

**Comment by perost** [comment]
> @sjoelund git blame shows a very old commit [111e614#diff-6f508873244f3e94c3614dcb4fca9b69709980cc190c597eca9f3c4878016e8d](https://github.com/OpenModelica/OpenModelica/commit/111e61493b87281bd2626754d20c27990b8a6c9f#diff-6f508873244f3e94c3614dcb4fca9b69709980cc190c597eca9f3c4878016e8d)

If you're wondering where that line was added it was actually added in [11467](https://trac.openmodelica.org/OpenModelica/changeset/11467).

**Comment by casella** [comment]
We should ask Jens why he did that 😄

**Comment by casella** [comment]
Waiting for @AnHeuermann's fixes

**Comment by casella** [comment]
Thanks @AnHeuermann!

@paultjevdh this is already available in the nightly builds and will be included in the 1.23.0 release, the first beta is due in one/two weeks max.

**Comment by SVAGEN26** [comment]
@JKRT requested I verify the status of this. Confirming it is resolved on current master (`v1.28.0-dev`).

Reproducer from the original report (the `test_space` model with `Real 'a x';` and `Real 'b_x';`) compiles and simulates cleanly. `readSimulationResultVars` on the result file lists both variables correctly:

```
{"'a x'", "'b_x'", "der('a x')", "der('b_x')", "time"}
```

So the underlying data that OMEdit's plot view consumes is correct. Fixed in 1.23.0 per the closing thread and still holds. No further action needed.
